pkg/aws/helpers.go

```diff
@@ -374,10 +374,27 @@ func GetPrefixFromAccountRole(cluster *cmv1.Cluster) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	rolePrefix := strings.TrimSuffix(roleName, fmt.Sprintf("-%s-Role", role.Name))
+	rolePrefix := TrimRoleSuffix(roleName, fmt.Sprintf("-%s-Role", role.Name))
 	return rolePrefix, nil
 }
 
+func GetPrefixFromOperatorRole(cluster *cmv1.Cluster) string {
+	operator := cluster.AWS().STS().OperatorIAMRoles()[0]
+	roleName := strings.SplitN(operator.RoleARN(), "/", 2)[1]
+	rolePrefix := TrimRoleSuffix(roleName, fmt.Sprintf("-%s-%s", operator.Namespace(), operator.Name()))
+	return rolePrefix
+}
+
+// Role names can be truncated if they are over 64 chars, so we need to make sure we aren't missing a truncated suffix
+func TrimRoleSuffix(orig, sufix string) string {
+	for i := len(sufix); i >= 0; i-- {
+		if strings.HasSuffix(orig, sufix[:i]) {
+			return orig[:len(orig)-i]
+		}
+	}
+	return orig
+}
```

Comment:
They would, yes. It just seems like this function would keep going until there are no characters left. So if the role ends in the same single character as the shortened suffix, then it would match?

Example:
```
TrimRoleSuffice("ManagedOpenShift-operator-openshift", "-openshift-ingress-operator-cloud-credentials")
```
it will keep truncating the suffix until only `-openshift` is left and so it would match a role that may have nothing to do with the ingress operator.

In any case, this seems like an unlikely scenario since we control the role names, but want to call it out as a possible risk.